### ISSUE TITLE
SelectCheckboxMenu: emptyLabel not considered as overlay-target and opens behind dialog

### ISSUE THREAD

**Issue by arnis87** [primefaces/primefaces#9988]
### Describe the bug

Dialog `bindEvents` checks for whether `mousedown` event target was `primefaces-overlay-target` and if not moves to top. This causes issues with clicks on `selectCheckboxMenu` `emptyLabel` which opens the menu/dropdown but dialog moves on top of it. When clicking on dropdown arrow or actual data label it works fine.

`SelectCheckboxMenu` `emptyLabel` fails this check on `line 379`:
https://github.com/primefaces/primefaces/blob/4e4430b40a2d1573137a975071edad1b3f953b66/primefaces/src/main/resources/META-INF/resources/primefaces/dialog/dialog.js#L377-L382
 

### Reproducer

Place selectCheckboxMenu inside popup and click on emptyLabel instead of dropdown arrow.
Menu opens but container dialog moves on top right after

### Expected behavior

Menu should open on top of the dialog

### PrimeFaces edition

Elite

### PrimeFaces version

12.0.3

### JSF implementation

Mojarra

### JSF version

2.3

### Java version

8

**Comment by melloware** [comment]
Please provide an executable example using the [PrimeFaces Test](https://github.com/primefaces/primefaces-test) project.  It is the only way developers can debug your problem to help.

**Comment by arnis87** [comment]
I will give it a go when I have more time. 

For now I have debugged it myself and can see that the issue is caused by the `emptyLabel` being rendered without every getting the `data('primefaces-overlay-target')` attached to it. During init it will `renderLabel` after the `data` is attached to `triggers` and its children.
`data('primefaces-overlay-target')` is attached here: https://github.com/primefaces/primefaces/blob/4e4430b40a2d1573137a975071edad1b3f953b66/primefaces/src/main/resources/META-INF/resources/primefaces/forms/forms.selectcheckboxmenu.js#L152

`renderLabel()` call happens few lines down.
https://github.com/primefaces/primefaces/blob/4e4430b40a2d1573137a975071edad1b3f953b66/primefaces/src/main/resources/META-INF/resources/primefaces/forms/forms.selectcheckboxmenu.js#L161
and the `emptyLabel` gets constructed inside via regular html injection and no data attached
https://github.com/primefaces/primefaces/blob/4e4430b40a2d1573137a975071edad1b3f953b66/primefaces/src/main/resources/META-INF/resources/primefaces/forms/forms.selectcheckboxmenu.js#L248

It is also issue when the `emptyLabel` has default value.

I have implemented a workaround/monkey patch which is to extend `PrimeFaces.widget.SelectCheckboxMenu` and its `renderLabel` method:


```
renderLabel: function () {
     this._super();
     this.triggers.data('primefaces-overlay-target', true).find('*').data('primefaces-overlay-target', true);
}
```
